### PR TITLE
Parameterize service thread pool impl

### PR DESCRIPTION
**Pull request by chwarr**

### CODE REVIEW

File: cpp/inc/bond/ext/grpc/thread_pool.h [hunk 1]
```diff
@@ -25,41 +25,61 @@
...
 namespace bond { namespace ext
 {
+
+#if 0
+/// @brief The interface that a compliant thread pool must implement.
+class thread_pool_concept
+{
+    /// @brief Schedules a callback for execution.
+    ///
+    /// @warning The scheduled callback must be exectured at some point in the
```
Comment (lalo) [inline cpp/inc/bond/ext/grpc/thread_pool.h]: typo: executed

File: cpp/inc/bond/ext/grpc/thread_pool.h [hunk 2]
```diff
@@ -25,41 +25,61 @@
...
+    /// @param numThreads total number of threads to be created. If
+    /// \ref USE_HARDWARE_CONC then as many threads as CPU/cores are available
+    /// will be created.
     explicit
-    thread_pool(uint32_t numThreads = 0)
+    thread_pool(size_t numThreads = USE_HARDWARE_CONC)
         : _work(*this)
     {
         //
         // If preferred # of threads is 0, use # of cpu cores.
```
Comment (lalo) [inline cpp/inc/bond/ext/grpc/thread_pool.h]: misplaced comment

File: cpp/test/grpc/thread_pool.cpp [hunk 3]
```diff
@@ -11,24 +11,22 @@
 // TODO: move unit_test_framework.h to cpp/test/inc
 #include "../core/unit_test_framework.h"
 
-#include <boost/chrono.hpp>
 #include <atomic>
+#include <chrono>
 #include <functional>
```
Comment (lalo) [inline cpp/test/grpc/thread_pool.cpp:16]: should this include thread_pool ?

**Review by lalo — CHANGES_REQUESTED** [review]

File: cpp/test/grpc/thread_pool.cpp (hunk 3 above)
Comment (chwarr) [reply cpp/test/grpc/thread_pool.cpp:16]: Already does, but a few system headers were missing. Fixed in commit 467ce0f.

**Review by lalo — APPROVED** [review]

**Comment by chwarr** [comment]
Merged into grpc++ in commit 7bbc2c034cb2d534b16f4b2d0e9b22ac6a5682d1

**Pull request closed without merging**